cli/cli.py

```diff
@@ -0,0 +1,155 @@
+"""
+CLI for interacting with the REMOTE analysis-runner
+    Motivation: https://github.com/populationgenomics/analysis-runner/issues/8
+
+Mostly for interacting with the server/main.py
+
+Called with:
+
+    analysis-runner run \
+        --output-dir gs://some-path/ \
+        --commit-hash <current-hash> \
+        --repo <current-repo>  \
+        --script <default: main.py>
+Process:
+
+    1. Fill in the mising info (repo, commit hash, etc)
+    2. Get the gcloud auth token $(gcloud auth print-identity-token)
+    3. Get the submit URL from the batch parameter
+    4. Form a POST request with params:
+        * output
+        * repo
+        * commit
+        * script
+        * description
+    5. Collect and print response
+"""
+from getpass import getuser
+
+import subprocess
+import click
+import requests
+import google.auth
+import google.auth.transport.requests
+
+BRANCH = 'add-cli'  # 'master'
+DEFAULT_STACK_LOOKUP = (
+    f'https://raw.githubusercontent.com/'
+    f'populationgenomics/analysis-runner/{BRANCH}/cli/stacklookup.json'
+)
+
+
+@click.command()
+@click.option(
+    '--stack',
```

Comment:
"Dataset" might be easier to understand for analysts, as that's the term we use in the storage policies. Using a Pulumi stack is just the concrete technical implementation.

I'm not super happy with "dataset" either, but "project" is already terribly overloaded. Let me know if you have a better idea :).

Reply:
I agree staying away from _project_, I think dataset is a reasonable middle-ground, hence I've renamed it to `dataset`.